march_hardware_builder/src/hardware_builder.cpp

```diff
@@ -257,3 +270,30 @@ void HardwareBuilder::initUrdf()
     this->init_urdf_ = false;
   }
 }
+
+std::string rightHandJustifyString(std::string input)
+{
+  size_t pos = 0;
+  size_t old_pos = 0;
+  std::string delimiter = "\n";
+  std::string token, result;
+  while ((pos = input.find(delimiter, old_pos)) != std::string::npos)
+  {
+    token = input.substr(old_pos, pos - old_pos);
+    if (token.length() > delimiter.length())
+    {
+      while (token.size() < 4 + delimiter.length())  // check until the 16-bit number has been filled
+      {
+        token.insert(0, "0");
+      }
+    }
+    result += input;
+    old_pos = pos + 1;  // Make sure to check the position after the previous one
+  }
+  return input;
+}
```

Comment:
Why is this method even necessary in the first place? Wouldn't the values `00AB` and `AB` be the same according to `std::stoi`? Furthermore, [`substr`](https://www.cplusplus.com/reference/string/string/substr/) returns a newly constructed string, so the `"0"` s you are inserting are discarded.

Reply:
You are right, it is no longer necessary. In previous versions it was useful for finding the previous token in the string because this made the string more predictable, but I have since moved on from that method and forgot to remove this function.